pkg/agent/route/route.go

```diff
@@ -310,10 +322,120 @@ func (c *Client) RemoveServiceRouting() error {
 	ipRule := netlink.NewRule()
 	ipRule.IifName = c.nodeConfig.GatewayConfig.Name
 	ipRule.Mark = iptables.RtTblSelectorValue
+	ipRule.Mask = iptables.RtTblSelectorValue
 	ipRule.Table = AntreaServiceTableIdx
 	ipRule.Priority = AntreaIPRulePriority
 	if err = netlink.RuleDel(ipRule); err != nil {
 		return fmt.Errorf("ip rule delete: %w", err)
 	}
 	return nil
 }
+
+// resolveDefaultRouteNHMAC resolves the MAC of default route next
+// hop on service route table.
+func (c *client) resolveDefaultRouteNHMAC() (net.HardwareAddr, error) {
+	// This MAC is relevant to L2 mode only.
+	// TODO, for now uses AKS default MAC.
+	return net.ParseMAC("12:34:56:78:9a:bc")
+}
+
+// SetupPolicyOnlyMode configures routing needed by traffic policy-only mode.
+func (c *client) SetupPolicyOnlyMode() error {
+	gwLink := util.GetNetLink(c.nodeConfig.GatewayConfig.Name)
+	_, gwIP, _ := net.ParseCIDR(fmt.Sprintf("%s/32", c.nodeConfig.NodeIPAddr.IP.String()))
```

Comment:
Sorry, I did not know we need an IP on gw0. But how could we know this IP is not used (e.g. reserved from underlay router)?

Reply:
perhaps already explained in earlier comment. The IP given to gw0 is exactly the same as Node IP. The reason is so that service traffic to a non-pod server can be masqueraded.

Reply:
Ok, I now see you add NodeIP to gw0 here. Still I do not know it can work - the same IP can be on two interfaces?

And could you add comments with details why we need to do this? It should be described in your design doc as well.